cuda/matrix/hybrid_kernels.cu

```diff
@@ -0,0 +1,436 @@
+    size_type grid_num = ceildiv(coo_num_stored_elements, default_block_size);
+    kernel::convert_coo_row_idxs_to_ptrs<<<grid_num, default_block_size>>>(
+        as_cuda_type(coo_row), coo_num_stored_elements,
+        as_cuda_type(coo_offset.get_data()), num_rows + 1);
+
+    // Compute the row ptrs of Csr
+    using cuda_size_type = unsigned long long int;
+    auto row_ptrs = result->get_row_ptrs();
+    auto coo_row_ptrs = Array<cuda_size_type>(exec, num_rows);
+
+    zero_array(num_rows + 1, row_ptrs);
+    grid_num = ceildiv(num_rows, warps_in_block);
+    kernel::count_ell_nnz_per_row<<<grid_num, default_block_size>>>(
+        num_rows, max_nnz_per_row, stride, as_cuda_type(ell_val),
+        as_cuda_type(row_ptrs));
+
+    zero_array(num_rows, coo_row_ptrs.get_data());
+
+    auto warps_per_sm = exec->get_num_cores_per_sm() / cuda_config::warp_size;
+    auto nwarps = host_kernel::calculate_nwarps(
+        coo_num_stored_elements, exec->get_num_multiprocessor() * warps_per_sm);
+    if (nwarps > 0) {
+        int num_lines =
+            ceildiv(coo_num_stored_elements, nwarps * cuda_config::warp_size);
+        const dim3 coo_block(cuda_config::warp_size, warps_in_block, 1);
+        const dim3 coo_grid(ceildiv(nwarps, warps_in_block), 1);
+
+        kernel::count_coo_row_nnz<<<coo_grid, coo_block>>>(
+            coo_num_stored_elements, num_lines, as_cuda_type(coo_val),
+            as_cuda_type(coo_row), as_cuda_type(coo_row_ptrs.get_data()));
+    }
+
+    kernel::add<<<grid_num, default_block_size>>>(
+        num_rows, as_cuda_type(row_ptrs),
+        as_cuda_type(coo_row_ptrs.get_const_data()));
+
+    grid_num = ceildiv(num_rows + 1, default_block_size);
+    auto add_values = Array<IndexType>(exec, grid_num);
+
+    start_prefix_sum<default_block_size>
+        <<<grid_num, default_block_size>>>(num_rows + 1, as_cuda_type(row_ptrs),
+                                           as_cuda_type(add_values.get_data()));
+
+    finalize_prefix_sum<default_block_size><<<grid_num, default_block_size>>>(
+        num_rows + 1, as_cuda_type(row_ptrs),
+        as_cuda_type(add_values.get_const_data()));
+
+    // Fill the value
+    grid_num = ceildiv(num_rows, default_block_size);
+    kernel::fill_in_csr<<<grid_num, default_block_size>>>(
+        num_rows, max_nnz_per_row, stride, as_cuda_type(ell_val),
+        as_cuda_type(ell_col), as_cuda_type(coo_val), as_cuda_type(coo_col),
+        as_cuda_type(coo_offset.get_const_data()), as_cuda_type(row_ptrs),
+        as_cuda_type(result->get_col_idxs()),
+        as_cuda_type(result->get_values()));
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_HYBRID_CONVERT_TO_CSR_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void count_nonzeros(std::shared_ptr<const CudaExecutor> exec,
+                    const matrix::Hybrid<ValueType, IndexType> *source,
+                    size_type *result)
+{
+    using cuda_size_type = unsigned long long int;
```

Comment:
You have this in a lot of places. Please use either `size_type`, or put a more general type in the `types.hpp` header.